src/transformers/generation_utils.py

```diff
@@ -1106,6 +1120,12 @@ def generate(
             # if decoder-only then inputs_tensor has to be `input_ids`
             input_ids = inputs_tensor
 
+        # Prepare exponential_decay_length_penalty, so decay start is applied to newly generated tokens
```

Comment:
Sorry, final change request. We are quite strict on not adding code that is to specific here. Could you instead add this code to `ExponentialDecayLengthPenalty` by passing the input_ids_shape as a parameter to init? 

This way all code specific to `ExponentialDecayLengthPenalty` is contained in this class

Reply:
Apart from this, the PR looks good to me :-)

Reply:
@patrickvonplaten sorry for the radio silence (was on vacation). I updated the PR :)

Reply:
@patrickvonplaten ping :)